tests/manage/pv_services/test_rbd_pv_encryption.py

```diff
@@ -0,0 +1,199 @@
+import logging
+import pytest
+
+from ocs_ci.framework.testlib import (
+    ManageTest,
+    tier1,
+    polarion_id,
+    skipif_ocs_version,
+)
+from ocs_ci.helpers.helpers import (
+    create_unique_resource_name,
+    create_pods,
+)
+from ocs_ci.ocs import constants, defaults
+from ocs_ci.ocs.exceptions import CommandFailed
+from ocs_ci.utility import kms
+from ocs_ci.ocs.ocp import OCP
+
+log = logging.getLogger(__name__)
+
+
+@skipif_ocs_version("<4.7")
+class TestRbdPvEncryption(ManageTest):
+    """
+    Test to verify RBD PV encryption
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        """
+        Setup csi-kms-connection-details configmap
+
+        """
+        # Initialize Vault
+        self.vault = kms.Vault()
+        self.vault.gather_init_vault_conf()
+        self.vault.update_vault_env_vars()
+
+        # Check if cert secrets already exist, if not create cert resources
+        ocp_obj = OCP(kind="secret", namespace=constants.OPENSHIFT_STORAGE_NAMESPACE)
+        try:
+            ocp_obj.get_resource(resource_name="ocs-kms-ca-secret", column="NAME")
+        except CommandFailed:
+            self.vault.create_ocs_vault_cert_resources()
+
+        # Create vault namespace, backend path and policy in vault
+        self.vault_resource_name = create_unique_resource_name("test", "vault")
+        self.vault.vault_create_namespace(namespace=self.vault_resource_name)
+        self.vault.vault_create_backend_path(backend_path=self.vault_resource_name)
+        self.vault.vault_create_policy(policy_name=self.vault_resource_name)
+
+        ocp_obj = OCP(kind="configmap", namespace=constants.OPENSHIFT_STORAGE_NAMESPACE)
+
+        # If csi-kms-connection-details exists, edit the configmap to add new vault config
+        try:
+            ocp_obj.get_resource(
+                resource_name="csi-kms-connection-details", column="NAME"
+            )
+            kmsid_list = kms.get_encryption_kmsid()
+            self.new_kmsid = f"{len(kmsid_list) + 1}-vault"
```

Comment:
can we always guarantee that next sequence number is always available ? I mean what if there are holes in the sequences because of creation and deletion from other tests and new_kmsid might collide. I am not sure about the tests hence asking.

Reply:
Good point. I have updated the new_kmsid to be a random value.